library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -745,8 +745,6 @@ simplifyConstraint' recurseIntoEvalBool = \case
         | isConstructorSymbol symbol = pure FalseBool
     evalEqualsK KSet{} (SymbolApplication symbol _ _)
         | isConstructorSymbol symbol = pure FalseBool
```

Comment:
This seems like it's probably not true?

What about two terms like `KSet(1, 2)` (internalized set), and `SetItem(1) |Set SetItem(2) |Set SetItem(2)`? These should be considered equal.

Reply:
I'm not familiar with the implementation - does `SymbolApplication` include things like `|Set`? In that case, could internalized maps and maps with map updates result in the same issue?

Reply:
sets are internalised, so there should be no possibility of KSet and a SymbolApplication being the same

Reply:
What about the above case though? Are symbols `_|Set_` internalized? It's a function, not a constructor. Do we have a test of this case?

Reply:
This case only applies to symbol applications of constructors. That's what the guard `| isConstructorSymbol symbol` is for. Since `_|Set_` is a function symbol, this case does not apply.